api/http/handler/auth/authenticate.go

```diff
@@ -123,21 +123,25 @@ func (handler *Handler) authenticateLDAPAndCreateUser(w http.ResponseWriter, use
 		return &httperror.HandlerError{http.StatusInternalServerError, "Unable to update user authorizations", err}
 	}
 
-	return handler.writeToken(w, user)
+	return handler.writeToken(w, user, userSessionTimeout)
 }
 
-func (handler *Handler) writeToken(w http.ResponseWriter, user *portainer.User) *httperror.HandlerError {
+func (handler *Handler) writeToken(w http.ResponseWriter, user *portainer.User, userSessionTimeout string) *httperror.HandlerError {
+	if userSessionTimeout == "" {
```

Comment:
This case should never happen, the Settings should have a default value. Use data migration to set the default value.